### ISSUE TITLE
Video: Allow streaming of all HEVC videos under Windows, e.g. by transcoding to AVC

### ISSUE THREAD

**Issue by metorm** [photoprism/photoprism#4067]
**Describe what problem this solves and why this would be valuable to many users**

There is a lot of discussion and controversy about what standards trigger transcoding and when. I believe that allowing users to choose to play the original file or the transcoded file on the page can solve many similar problems.

The specific problem I encountered is that some hevc/mov files shot by iPhone play normally on windows, and the avc files generated by transcoding after uploading play normally in browsers without hevc support. However, in Windows browsers with HEVC support, the system automatically selects the HEVC data stream, but it cannot be played due to some unknown reasons.

In fact, I don't care what kind of data stream I'm watching, I just want to be able to preview the file normally. However, since my browser has a HEVC decoder, I lost the ability to view the AVC stream. This is unreasonable.

**Describe the solution you'd like**

I wish there was an additional option on the webpage to watch the transcoded video.

**Describe alternatives you've considered**

Alternatively, there can be a global option in the system to control whether to provide transcoded video to all clients.

**Comment by lastzero** [comment]
This should not be happening and could be related to either your browser or [graphics driver](https://support.microsoft.com/en-us/windows/display-requirements-for-hdr-video-in-windows-192f362e-1245-e14d-3d3f-4b3fc606b80f). It might also be an issue with the specific HEVC file, as our video format support detection does not include checks for optional features like [HDR10 and HDR10+](https://en.wikipedia.org/wiki/High-dynamic-range_television):

https://github.com/photoprism/photoprism/blob/5bd822e616d8ee9e503f9670b009543a8b6f3e56/frontend/src/common/caniuse.js#L45-L47

So instead of investing time in a workaround, it would be great to get help improving those three lines of code (or our troubleshooting guide if it's a browser/driver issue that can be diagnosed and fixed). Thank you very much! 🙏

**Comment by metorm** [comment]
OK. My problem may have been caused by a cracked version of a HEVC UWP decoder. I'll test it later.

**Comment by metorm** [comment]
No, I think it's not a problem in the front end, but something in the core.

I tested more videos, and it turned out only those shot by iPhone triggered the issue. Those taken by Android phones are OK.

The network monitor indicates that when playing a hevc-encoded mp4 file, the browser gets a normal hevc stream like below:

![image](https://github.com/photoprism/photoprism/assets/12082680/a052b3ed-a33b-4a21-b185-02374172dfdb)

However, when I click a hevc-encoded mov file shot by iPhone, I got this in the network monitor:

![image](https://github.com/photoprism/photoprism/assets/12082680/1c7a0bab-3d18-4a29-8712-15d4337b28d0)

```
no data found for resource with given identifier
```

That's all I can found. May it be caused by a failed file parsing or something similar?

**Comment by lastzero** [comment]
To force the transcoding of non-AVC videos that are currently being streamed as original, a workaround could be to set a lower bitrate limit in your configuration, as HEVC-encoded videos with special features such as HDR/HFR generally have a high bitrate (despite the more modern codec).

In our upcoming release, the `hvc1.1.6.L93.90` part in the code snippet above could be changed to (also) check support for any optional features (like HDR10) so that such HEVC videos are not streamed to Windows, but only to macOS/iOS for as long as only these support them. Any help with this is much appreciated! ❤️

**Comment by lastzero** [comment]
## Update: HEVC sample-entry tag normalization landed (2026-05-20)

The "HEVC won't play in a Windows browser that has a HEVC decoder" symptom in this report is almost always caused by the source file being tagged `hev1` rather than `hvc1` — Edge and Chrome on Windows (and Safari on macOS / iOS) only decode the `hvc1` form. PhotoPrism's remux pipeline previously preserved whatever tag the source had, so HEVC files captured by some iPhone configurations or downloaded via the `dl` workbench inherited `hev1` and failed to play after import.

The fix landed in `12dbe82d3` — see #5593 for the full details. In short:

- `RemuxFile` auto-detects an HEVC source via the new `video.IsHEVCFile` helper and passes `-tag:v hvc1` to ffmpeg so MP4 / MOV outputs always carry the `hvc1` sample-entry code.
- `Probe()` now recognizes all eight HEVC variants (`hvc1`, `hvc2`, `hvc3`, `dvh1`, `hev1`, `hev2`, `hev3`, `dvhe`) in a single 16 MiB head scan and normalizes through the existing `Codecs` table; Dolby Vision HEVC wrappers (`dvh1`, `dvhe`) that were previously reported as `CodecUnknown` are now classified correctly.
- HEVC chunk scans are bounded (no more end-to-end reads of GB-sized non-HEVC files).

### Remaining gaps

1. **Pre-existing `hev1`-tagged files in user libraries** are not retroactively rewritten. Run `photoprism remux --force` once to re-tag them; the command lifts the AVC-only gate when `--force` is set and now emits `hvc1` via the new auto-detect. The command's `videoRemuxFile` path already re-indexes after the rewrite, so the DB-cached `FileCodec` and the served `Content-Type` follow.
2. **Manual file rewrites need a re-index.** `file.FileCodec` is set from exiftool output at index time; if a user re-tags an HEVC file outside the `remux` CLI, the api keeps advertising the stale `codecs="hev1.*"` until the next index pass.
3. **No automatic re-tag at index time.** An opt-in indexer hook that calls `RemuxFile` when it encounters an `hev1`-tagged file would close this gap without a manual step. Out of scope for the current change; happy to track as a separate enhancement if it's worth doing.

For deeper context (worst-case work bounds, caller survey, manual ffprobe verification of a real HEVC remux), see #5593.
